File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;
int answer = 0;
vector<int> n;

int t;
void dfs(int cnt, int cur){
    if(cnt == n.size()){
        if(t == cur) {answer++;}
        
        return;
    }
    dfs(cnt+1, cur+n[cnt]);
    dfs(cnt+1, cur-n[cnt]);
    
}

int solution(vector<int> numbers, int target) {
    n = numbers;
    t = target;
    dfs(0,0);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
let tg = 0;
let nums = [];
let answer = 0;

function dfs(cur, index, debug) {
    if(index == nums.length - 1){
        // console.log("ㅠㅠ")
        if(cur == tg){
        // console.log("야호 타겟이다")
        answer++;
        return;
    }
        return;
    }

    dfs(cur+nums[index+1], index+1, `${debug} + "+" + ${nums[index+1]}`)
    dfs(cur-nums[index+1], index+1, `${debug} + "-" + ${nums[index+1]}`)
    
}

function solution(numbers, target) {
    tg = target;
    nums = numbers;
    
    dfs(numbers[0], 0, numbers[0]);
    dfs(-numbers[0], 0, -numbers[0]);
    
    return answer;
}